src/core/segment_reader.rs

```diff
@@ -69,6 +70,18 @@ impl SegmentReader {
         &self.schema
     }
 
+    /// Merges the passed bitset with the existing one.
+    pub fn apply_alive_bitset(&mut self, alive_bitset: AliveBitSet) -> crate::Result<()> {
```

Comment:
I'm really scared of code that mutates SegmentReader.
Is it possible to have the following method prototype:

```
pub fn apply_alive_bitset(&self, alive_bitset: AliveBitSet) -> crate::Result<SegmentReader>
```

Reply:
I moved it to the constructor